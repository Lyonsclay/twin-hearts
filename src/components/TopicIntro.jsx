import React, { Component } from 'react'
import MediaQuery from 'react-responsive'

export const styles = {
  main: {
    marginBottom: "-4em"
  },
  intro: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-around'
  },
  introMobile: {
    display: 'flex',
    flexDirection: 'column-reverse',
    justifyContent: 'space-around'
  },
  details: {
    overflow: 'hidden',
    flex: '0 0 auto',
    contentJustify: 'space-around',
    flexBasis: '30%',
    textAlign: 'center',
    minWidth: 240
  },
  poster: {
    marginLeft: 20,
    marginRight: 20,
    height: 450,
    textAlign: 'center',
    minWidth: 480
  },
  title: {
    marginTop: 15
  },
  description: {
    display: 'flex',
    flexGrow: 1,
    flexDirection: 'column',
    textAlign: 'center',
    margin: 40
  },
  thumbs: {
    height: 90
  },
  img: {
    margin: 15,
    maxWidth: '100%',
    maxHeight: '100%'
  }
};


export const Poster = ({url}) => {
  return (
    <div style={styles.poster}>
      <img src={url} style={styles.img} alt="poster" />
    </div>
  );
};

export const Title = ({title}) => {
  return (
    <div style={styles.title}>
      <h1>{title}</h1>
    </div>
  );
};

export const Description = ({description}) => {
  return (
    <div style={styles.description}>
      <p>{description}</p>
    </div>
  );
};

export class Thumbnails extends Component {
  render() {
    let images = []
    try {
      images = this.props.thumbnails.map((url, index) => (
        <img src={url} key={index} style={styles.img} alt="thumb" />
      ));
    } catch (e) {
      console.log('Error receiving Thumbnails', e)
    }

    return (
      <div style={styles.thumbs}>
        {images}
      </div>
    )
  }
}

export const Details = ({title, description, screenshots}) => {
  return (
    <div style={styles.details}>
      <Title title={title} />
      <Description description={description} />
      <Thumbnails thumbnails={screenshots} />
    </div>
  );
};

const TopicIntro = (props) => {
  return (
    <div style={styles.main}>
      <MediaQuery minWidth={1100} >
        <div style={styles.intro}>
          <Details {...props} />
          <Poster url={props.image} />
        </div>
      </MediaQuery>
      <MediaQuery maxWidth={1100} >
        <div style={styles.introMobile}>
          <Details {...props} />
          <Poster url={props.image} />
        </div>
      </MediaQuery>
    </div>
  );
};

export default TopicIntro;
