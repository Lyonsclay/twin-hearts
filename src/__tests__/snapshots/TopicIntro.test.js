import React from 'react'
import renderer from 'react-test-renderer'
import TopicIntro from '../../components/TopicIntro'

describe('TopicIntro', () => {
  const props = {
    title: 'Great Show',
    description: 'It rips the curtain off the American dream!',
    poster: 'http://thetvdb.com/banners/fanart/original/28950-23.jpg',
    screenshots: [
      "http://thetvdb.com/banners/fanart/original/289590-25.jpg",
      "http://thetvdb.com/banners/fanart/original/289590-6.jpg",
      "http://thetvdb.com/banners/fanart/original/289590-8.jpg"
    ]
  }

  const intro = renderer.create(
    <TopicIntro {...props} />
  )

  let tree = intro.toJSON()

  it('renders', () => {
    expect(tree).toMatchSnapshot();
  })
})
