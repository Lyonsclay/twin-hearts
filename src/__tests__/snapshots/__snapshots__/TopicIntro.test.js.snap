exports[`TopicIntro renders 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-around"
    }
  }>
  <div
    style={
      Object {
        "contentJustify": "space-around",
        "flex": "0 0 auto",
        "flexBasis": "30%",
        "minWidth": 240,
        "overflow": "hidden",
        "textAlign": "center"
      }
    }>
    <div
      style={
        Object {
          "marginTop": 15
        }
      }>
      <h1>
        Great Show
      </h1>
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "flexGrow": 1,
          "margin": 40,
          "textAlign": "center"
        }
      }>
      <p>
        It rips the curtain off the American dream!
      </p>
    </div>
    <div
      style={
        Object {
          "height": 90
        }
      }>
      <img
        src="http://thetvdb.com/banners/fanart/original/289590-25.jpg"
        style={
          Object {
            "margin": 15,
            "maxHeight": "100%",
            "maxWidth": "100%"
          }
        } />
      <img
        src="http://thetvdb.com/banners/fanart/original/289590-6.jpg"
        style={
          Object {
            "margin": 15,
            "maxHeight": "100%",
            "maxWidth": "100%"
          }
        } />
      <img
        src="http://thetvdb.com/banners/fanart/original/289590-8.jpg"
        style={
          Object {
            "margin": 15,
            "maxHeight": "100%",
            "maxWidth": "100%"
          }
        } />
    </div>
  </div>
  <div
    style={
      Object {
        "height": 500,
        "marginLeft": 20,
        "marginRight": 20,
        "minWidth": 480,
        "textAlign": "center"
      }
    }>
    <img
      src={undefined}
      style={
        Object {
          "margin": 15,
          "maxHeight": "100%",
          "maxWidth": "100%"
        }
      } />
  </div>
</div>
`;
